### ISSUE TITLE
Moment overwrites itself and all extensions

### ISSUE THREAD

**Issue by DrkCloudStrife** [moment/moment#5363]
**Describe the bug**
Apologies in advance if this belongs in moment timezone. When working on a large application, there are a lot of dependencies that are loaded through various means. In this case, a plugin that depends on moment is loaded to a large application that uses moment and moment timezone. The plugin only requires moment, but is loading moment with itself. When this happens, the extensions on the original moment are overwritten. See this jsfiddle for example https://jsfiddle.net/uth41pnk/1/

**To Reproduce**
1. Create an application
2. Load moment.js
3. Load moment.js timezone
4. Load a plugin that depends and includes moment.js
5. Use moment.js timezone after the plugin has been loaded

**Expected behavior**
I should expect `moment.tz` to be defined and work as intended.

**Screenshots**
![image](https://user-images.githubusercontent.com/98049/73096184-dd373780-3ea9-11ea-8be4-aa18a98bdb4f.png)

**Moment-specific environment**

* The time zone setting of the machine the code is running on
* The time and date at which the code was run
* Other libraries in use (TypeScript, Immutable.js, etc)

Please run the following code in your environment and include the output:
```
Fri Jan 24 2020 13:04:18 GMT-0600 (Central Standard Time)
1/24/2020, 1:04:18 PM
360
Mozilla/5.0 (Macintosh; Intel Mac OS X 10_14_6) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/79.0.3945.117 Safari/537.36
2.10.2
```

**Additional context**
While I know we are currently working on a very old version of moment (2.10.2), I would expect moment to check if it has already been defined. Currently, I have monkey patched the plugin not to include moment.

**Comment by Alanscut** [comment]
@DrkCloudStrife You should import `moment-timezone`. See https://momentjs.com/timezone/.

**Comment by DrkCloudStrife** [comment]
@Alanscut Thanks for the reference. In the steps I provided I stated I'm using `moment-timezone`, albeit I wrote it as `moment.js timezone`. 

To give more context, I'm not using webpack and I'm using a rails application with sprockets 3. The problem is with a third party plugin that depends on `moment.js`. This plugin is overwriting my `moment.js` configuration. What I'm asking is for `moment.js` to detect if it has already been defined prior to defining itself. Similar to how polyfills behave, so that the original pre-defined class isn't overwritten.
